lib/malloc.cpp

```diff
@@ -11,38 +11,53 @@
 #include "qdp_config.h"
 #endif
 
-namespace quda {
+#ifdef QUDA_BACKWARDSCPP
+#include "backward.hpp"
+#endif
+namespace quda
+{
 
   enum AllocType { DEVICE, DEVICE_PINNED, HOST, PINNED, MAPPED, MANAGED, N_ALLOC_TYPE };
 
-  class MemAlloc {
+  class MemAlloc
+  {
 
   public:
     std::string func;
     std::string file;
     int line;
     size_t size;
     size_t base_size;
+#ifdef QUDA_BACKWARDSCPP
+    backward::StackTrace st;
+#endif
 
-    MemAlloc()
-      : line(-1), size(0), base_size(0) { }
+    MemAlloc() : line(-1), size(0), base_size(0) {}
 
-    MemAlloc(std::string func, std::string file, int line)
-      : func(func), file(file), line(line), size(0), base_size(0) { }
+    MemAlloc(std::string func, std::string file, int line) : func(func), file(file), line(line), size(0), base_size(0)
+    {
+#ifdef QUDA_BACKWARDSCPP
+      st.load_here(32);
```

Comment:
curious: what's the significance of the 32?

Reply:
It is the max depth of the trace. From the documentation:
```	
        // Take a snapshot of the current stack, with at most "trace_cnt_max"
	// traces in it. The first trace is the most recent (ie the current
	// frame). You can also provide a trace address to load_from() assuming
	// the address is a valid stack frame (useful for signal handling traces).
	// Both function return size().
	size_t load_here(size_t trace_cnt_max)
```

Tbh it was the default in all the examples and I never bother deviating from that.